### PR TITLE
feat: make prometheus v3.0.1 as default version

### PR DESCRIPTION
**Pull request by slashpai**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [x] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
feat: make prometheus v3.0.1 as default version
```

### CODE REVIEW

File: Documentation/compatibility.md [hunk 1]
```diff
@@ -53,18 +53,18 @@ Prometheus Operator supports all Prometheus versions >= v2.0.0. The operator's e
 * v2.54.1
 * v2.55.0
 * v2.55.1
+* v3.0.0
+* v3.0.1
 * v3.0.0-beta.0
```
Comment (simonpasquier) [inline Documentation/compatibility.md:58]: I would remove the beta/rc versions.

**Comment by simonpasquier** [comment]
Could we also lay out the changes regarding the e2e tests? E.g. adding a periodic job which would run the e2e tests against the latest v2.x release.

**Comment by slashpai** [comment]
> Could we also lay out the changes regarding the e2e tests? E.g. adding a periodic job which would run the e2e tests against the latest v2.x release.

Did you mean to update actions also part of this PR or mention about it in comments in this PR?

File: pkg/operator/defaults.go [hunk 2]
```diff
@@ -75,17 +79,19 @@ var (
...
+func getLatestPrometheusV2() string {
+	v2 := regexp.MustCompile(`v2`)
+	for i, version := range PrometheusCompatibilityMatrix {
+		match := v2.FindStringIndex(version)
+		// Since first non match would a v3 version
+		if match == nil {
+			return PrometheusCompatibilityMatrix[i-1]
+		}
+	}
+	return ""
```
Comment (simonpasquier) [inline pkg/operator/defaults.go:96]: (nit) we can fail hard.
```suggestion
	panic("failed to find a v2.x entry in the compatibility matrix")
```

File: pkg/operator/defaults.go [hunk 3]
```diff
@@ -75,17 +79,19 @@ var (
...
-		"v3.0.0-rc.0",
-		"v3.0.0-rc.1",
-		// TODO: To be moved to default latest version once we tested fully.
 		"v3.0.0",
 		"v3.0.1",
 	}
 )
+
+func getLatestPrometheusV2() string {
+	v2 := regexp.MustCompile(`v2`)
```
Comment (simonpasquier) [inline pkg/operator/defaults.go:88]: (nit) we can use https://pkg.go.dev/strings#HasPrefix

**Review by simonpasquier — COMMENTED** [review]
Thanks for updating the GH actions too!

**Review by simonpasquier — APPROVED** [review]
thanks!

**Pull request merged**